impl/src/main/java/org/jboss/weld/injection/InterceptionFactoryImpl.java

```diff
@@ -137,4 +169,25 @@ public T createInterceptedInstance(T instance) {
         return proxy;
     }
 
+    private boolean isInterceptorBinding(Annotation annotation) {
+        return annotation.annotationType().getAnnotation(InterceptorBinding.class) != null;
+    }
+
+    private boolean isMatchingMethod(Method first, Method second) {
+        if (first == null || second == null) {
+            return false;
+        }
+        if (!first.getName().equals(second.getName())
+            || !first.getReturnType().equals(second.getReturnType())
+            || !(first.getParameterCount() == second.getParameterCount())) {
+            return false;
+        }
+        List<Class<?>> secondMethodParams = Arrays.asList(second.getParameterTypes());
+        for (Class<?> paramType : first.getParameterTypes()) {
```

Comment:
This would not work correctly for methods with generic params, or? Maybe `Method.getGenericParameterTypes()`?

Reply:
Not sure, will try to put together a test to prove that (and add it here).